lib/livebook/file_system/s3/client.ex

```diff
@@ -252,6 +252,11 @@ defmodule Livebook.FileSystem.S3.Client do
   end
 
   defp build_url(file_system, path, query) do
+    query =
+      if file_system.session_token,
+        do: Map.put(query, :"X-Amz-Security-Token", file_system.session_token),
```

Comment:
Do we need to pass this both in the signature and in the query string?

Reply:
Ah, sorry, I meant to change this to use the header instead of the query parameter. It does need to be one or the other.